python/common/org/python/types/List.java

```diff
@@ -167,8 +168,14 @@ public List(org.python.Object [] args, java.util.Map<java.lang.String, org.pytho
             int count = Math.min(size, otherSize);
 
             for (int i = 0; i < count; i++) {
-                org.python.types.Bool b =
-                    (org.python.types.Bool) this.value.get(i).__lt__(otherList.value.get(i));
+                org.python.Object result = this.value.get(i).__lt__(otherList.value.get(i));
```

Comment:
Sorry, I'm not too sure what you mean by `compare` here. Do you mean the logic implemented in the ast? If so, I think there's no nice way to invoke that since there is no longer a `__cmp__()` Java method.